packages/astro/src/runtime/server/render/astro/render.ts

```diff
@@ -100,13 +61,87 @@ export async function renderToReadableStream(
 					}
 
 					const bytes = chunkToByteArray(result, chunk);
-					controller.enqueue(bytes);
+					handler.push(bytes);
```

Comment:
Perf-wise, I'm a bit concerned that we're consolidating on handling plain `UInt8Array` here. `chunkToByteArray` and `chunkToString` are implemented this way:

https://github.com/withastro/astro/blob/47b951b3888a5a8a708d2f9b974f12fba7ec9ed3/packages/astro/src/runtime/server/render/common.ts#L131-L142

https://github.com/withastro/astro/blob/47b951b3888a5a8a708d2f9b974f12fba7ec9ed3/packages/astro/src/runtime/server/render/common.ts#L123-L129

Which means that with this PR, `renderToString`'s `string` chunks would convert to `UInt8Array`, and then back as `string` again:

https://github.com/withastro/astro/blob/6b4f77c100da3401f1749184e15b5218c947c70c/packages/astro/src/runtime/server/render/astro/render.ts#L119

It's the reason why the initial render functions has slightly duplicated code, because strings can be fast-path-ed. Perhaps we should keep the fast path like before?

Reply:
Yeah totally. I didn't feel great about that either. The reason I did it was because the `renderToString`, `renderToReadableStream` (and the new) `renderToAsyncIterator` are 90% duplicate code and wanted to avoid having to add any new code in 3 places. So this refactor was meant to make it so that the functions only handle their different outputs. But for `renderToString` I could either revert back to the old way, or figure out some way to make the abstraction handle either the typed array way or the string way.